flink-streaming-java/src/main/java/org/apache/flink/streaming/api/graph/StreamConfig.java

```diff
@@ -176,10 +175,18 @@ public CompletableFuture<StreamConfig> triggerSerializationAndReturnFuture(
                                                                 Function.identity())),
                                         this.config,
                                         CHAINED_TASK_CONFIG);
-                            } catch (IOException e) {
-                                throw new StreamTaskException(
-                                        "Could not serialize object for key chained task config.",
-                                        e);
+                            } catch (IOException | StreamTaskException e) {
+                                StreamTaskException serializationException;
+                                if (e instanceof IOException) {
+                                    serializationException =
+                                            new StreamTaskException(
+                                                    "Could not serialize object for key chained task config.",
+                                                    e);
+                                } else {
+                                    serializationException = (StreamTaskException) e;
+                                }
+                                serializationFuture.completeExceptionally(serializationException);
+                                throw serializationException;
```

Comment:
There's no point of throwing an exception in this thread. I think we can simply catch any throwable and complete the future with them.

Reply:
ok, fixed.